src/Package/Impl/DataInspect/Viewers/DebugObjectEvaluator.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.ComponentModel.Composition;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Common.Core;
+using Microsoft.R.Debugger;
+using Microsoft.R.Host.Client;
+using Microsoft.VisualStudio.R.Package.Repl;
+
+namespace Microsoft.VisualStudio.R.Package.DataInspect.Viewers {
+    [Export(typeof(IDebugObjectEvaluator))]
+    public sealed class DebugObjectEvaluator : IDebugObjectEvaluator {
+        private readonly IDebugSessionProvider _debugSessionProvider;
+        private readonly IRSessionProvider _rSessionProvider;
+        private DebugSession _debugSession;
+        private IRSession _rSession;
+
+        [ImportingConstructor]
+        public DebugObjectEvaluator(IRSessionProvider rSessionProvider, IDebugSessionProvider debugSessionProvider) {
+            _rSessionProvider = rSessionProvider;
+            _debugSessionProvider = debugSessionProvider;
+        }
+
+        public async Task<DebugEvaluationResult> EvaluateAsync(string expression, DebugEvaluationResultFields fields) {
+            await TaskUtilities.SwitchToBackgroundThread();
+            DebugEvaluationResult result = null;
+
+            var debugSession = await GetDebugSessionAsync();
+            var frames = await debugSession.GetStackFramesAsync();
+
+            for (int i = frames.Count - 1; i >= 0; i--) {
```

Comment:
I did basically this
```x <- mtcars```
inside deeper frame and View(x) from command line didn't work since it needed deeper frame rather than .GlobalEnv